### ISSUE TITLE
Add new "Stack View" for viewing open tabs

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#22028]
## Description
When using Brave, users can see the number of open tabs at the bottom of the browser
![image](https://user-images.githubusercontent.com/4733304/161160050-ddab23c2-c541-466f-8cd5-b02183de4fe7.png)

When clicking this, Brave will show the open tabs so users can see them. 

### Current behavior
This is **currently** done in a "Grid View" which looks like this:
<img src="https://user-images.githubusercontent.com/4733304/161160308-77fb0ad5-4409-48cf-8edf-09a7da8602cb.png" width="300px">

### Desired behavior
There should be a new configuration setting (Settings > Display > Appearance) which allows the user to change the view they see when viewing tabs. Options could be
- Grid View (default)
- Stack View

Here's an example of what Stack View would look like
<img src="https://user-images.githubusercontent.com/4733304/161159782-d5f8d5a1-99d4-4f90-92ba-904df1d1fcbc.png" width="300px">

Basically, all the open tabs are stacked, oldest at top and newest at bottom. The user can scroll through these vertically and see the cards peek as they're being shuffled. Tabs that need to be closed can be swiped off the screen to close the tab

cc: @anthonypkeane for triage

**Comment by futurejohn** [comment]
Yes, this is exactly what we want! There's also the feature where pressing and holding on a tab will cause the other tabs to slide away so that more of the current tab can be seen.

**Comment by picks-1** [comment]
If the Kiwi browser team can make it work, you should be able to. 
If you want to retain users, you need to put in effort where it's most needed. Excuses and complaints about "having to do a lot of work" won't retain users. 
I hope your prioritization is well considered.

**Comment by RowanSkie** [comment]
Do we have a visible timeframe for this? Because a lot of us have started moving away from Brave.

**Comment by Abscissa** [comment]
> It’s tempting to just add this feature back in, but we really need to consider the long term ownership of this and define some acceptance criteria. If we are going to add this feature back in, I believe we should do it properly, which will take some time.

Honestly, I *love* that this is being legitimately and sincerely worked on. But in all truthfulness, IMHO (FWIW), the *correct* high-level decision here is to revert the PR/UI-disaster change asap, and *THEN* work on fixing what you already admitted in the earlier (now closed) thread is only about a ~1-2% pre-existing technical issue.

Put out the fire, *THEN* fix the cause of the fire. Not the other way around.

**Comment by emrys767** [comment]
I'm grateful this is being considered, as I am sticking with an old beta build of Brave for now.  I'm hoping to see this issue included in next version of Brave.  I cannot stay on a beta build forever.

**Comment by bw4518** [comment]
What is the latest version with stack view, for the future reference.

**Comment by bsclifton** [comment]
@bw4518 it would be any version of Brave with Chromium 98. I believe that would be 1.35 or earlier

**Comment by ChrislyBear-GH** [comment]
The tab switching pane always mimicked the native Android app switching. What about the new app switching style for tabs?
Would it be easier to hook into this API and use it for tab switching? I mean, at this point anything would be better than a grid.

**Comment by JohnCulver** [comment]
> the _correct_ high-level decision here is to revert the PR/UI-disaster change asap

Agree with this. I have no intention of picking the app up again until this is resolved, and leaving a 1 star review in the app store because this really destroys the basic workflow of using a browser on a phone.

**Comment by rkreilly** [comment]
Is there any estimate on the time frame for this? Stack/List layout is so much more efficient and easy to use. Absolutely stupid to remove it.

It's incredibly annoying when the AOL-style layout messes up constantly. 75% of the time (or more) the damn thing tries to combine tabs or move them around, INSTEAD OF SIMPLY SCROLLING. Half the time the only way I can scroll through tabs is by creating a new tab to get to the bottom, reopen tab view. I'd another new tab if the last one was even. Reopen tab view. Use the empty space next to the odd tab to get scrolling moving, then I can keep scrolling instead of moving/combining tabs...

**Comment by Joshuffle** [comment]
Hope it gets readded but better than ever (aka that you can rearrange tabs with stack view aswell)

**Comment by Goranaws** [comment]
This needs to happen soon!! I am trapped on a version from December 2021, and I will not update until stack view is restored. It makes absolutely no sense to have removed it in the first place. Grid view is entirely unusable, and I can't comprehend why anyone would find it superior in any way.

**Comment by Goranaws** [comment]
> > This needs to happen soon!! I am trapped on a version from December 2021, and I will not update until stack view is restored. It makes absolutely no sense to have removed it in the first place. Grid view is entirely unusable, and I can't comprehend why anyone would find it superior in any way.
> 
> Because apparently, Chromium (and most likely Google) thought that smartphones are expanding to tablet size... and apparently it makes the apps crash on Android 12 so instead of improving or rewriting it, they removed it.

That's bad logic then, I have a laptop for that. I will never own a phone that doesn't fit in my pocket.

**Comment by ArmandFrvr** [comment]
This needs to be priority #1.  I've stuck it out for a month trying to get used to the new view but it's just completely untenable.

**Comment by RowanSkie** [comment]
Does anyone have any new news for this?

Also, any alternatives to Brave that has roughly the same features? Because as much as I want to wait, this feature is still better than the current one we have right now.

**Comment by JohnCulver** [comment]
> Also, any alternatives to Brave that has roughly the same features?

Kiwi is currently what I'm using while we wait and see what happens here. I believe another member in this thread also mentioned it.

**Comment by H4zh4n** [comment]
@RowanSkie 
> any alternatives to Brave that has roughly the same features?

This Stack View issue is genuinely annoying me on daily basis. now with brave's newest update comes another issue that the URL bar keeps hiding with no apparent reason (don't know if its only me).
**Kiwi Browser** seems to have a lot of Brave features (with Stack View Tabs), it supports Chrome Extensions (Real PC browser Chrome Store Extensions) which you can download all sort of stuffs (AdBlock, Dark Reader... Etc.).
The only thing that prevents me from fully switching over is they don't have a PC side browser that supports syncing between both, I use both my phone and PC daily and I want my bookmarks, history and stuff on both sides. If you are only working on your Phone I suggest you give it a try.

let's hope Brave's lovely team soon fixes and implements what the community wants.

**Comment by DrWhargoul** [comment]
> Does anyone have any new news for this?
> 
> Also, any alternatives to Brave that has roughly the same features? Because as much as I want to wait, this feature is still better than the current one we have right now.

OperaGX is what I'm using.

**Comment by Taomyn** [comment]
> now with brave's newest update comes another issue that the URL bar keeps hiding with no apparent reason (don't know if its only me). 

It's not just you https://community.brave.com/t/keep-losing-address-bar/384066

**Comment by RowanSkie** [comment]
> > now with brave's newest update comes another issue that the URL bar keeps hiding with no apparent reason (don't know if its only me).
> 
> It's not just you https://community.brave.com/t/keep-losing-address-bar/384066

Glad I've decided to go Chromium 98 Brave until the end then. Might try Kiwi Browser though.

**Comment by santacriptus** [comment]
I tried to survive with Brave 1.35.103 installed from apk and with Kiwi as the second browser. However after enough long negative experience of switching browsers I am going to uninstall Brave and to use Kiwi as my default and only browser.
I will still continue waiting for your fix of this issue, as well I still enjoy using Brave as one of my Ubuntu desktop browsers.
Thank you for your efforts.

**Comment by BringBackCardTabs** [comment]
Sick of waiting, ditching for Kiwi today.

**Comment by Abscissa** [comment]
I finally gave up in favor if Kiwi, too. It has an ad-blocking feature like Brave, plus an option for "stack view" tabs, and on top of that, even has options to make the address bar finally quit auto-hiding and also to finally stop harassing me about "simplified view" (or whatever its called). Those last two were things that had been annoying the hell out of me, but I didn't know I even COULD go without those irritants until I heard about Kiwi here and tried it.

**Comment by futurejohn** [comment]
> I finally gave up in favor of Kiwi, too. It has an ad-blocking feature like Brave, plus an option for "stack view" tabs, and on top of that, even has options to make the address bar finally quit auto-hiding and also to finally stop harassing me about "simplified view" (or whatever its called). Those last two were things that had been annoying the hell out of me, but I didn't know I even COULD go without those irritants until I heard about Kiwi here and tried it.

Same here. Kiwi has all the features of Brave and it's a bit more customizable. It even has a quick toggle for Night Mode that doesn't have to restart the whole browser. There's really no reason for me to switch back to Brave even if they do add stacked tabs back.

**Comment by Hoho5000** [comment]
I'm done with Brave as well. I've been a long time user and prefer Brave over other mobile options. I don't normally have a lot of tabs open, but when I do, the current forced layout is a mess.

Due to the lack of communication or priority on this, I'm switching to Kiwi and not coming back. This should not still be an unstarted issue almost 3 months later.

**Comment by theridon** [comment]
I've been using Kiwi browser now, works the same as Brave but has a tab layout that isn't complete garbage. At this point it's obvious Brave devs are not interested in giving us the old tabs, I just uninstalled the Brave app today and will be using Kiwi exclusively.

**Comment by RowanSkie** [comment]
@bsclifton got any comments? Because people are leaving Brave for Kiwi browser now. 

And then there's also that bug which was found because people want to use this.

People are still waiting for this, you know.

**Comment by bsclifton** [comment]
Hi folks - yes; this is prioritized and we'll start work in the next few weeks. We are committed to this

I'm going to lock this issue and clean things up a bit. I definitely appreciate how passionate folks are - thanks for your patience

**Comment by bsclifton** [comment]
+1 from @notvenom via https://github.com/brave/brave-browser/issues/26393

**Comment by Enriador** [comment]
> we are pushing through updates to rewards, news, vpn and privacy hub first.

Thanks for the news, after 5+ months stuck on the Beta 1.36 version this is good to hear - I miss the new features yet grid tab layout causes me physical pain due to severe hand strain.

It is, however, sort of disturbing how things like "rewards" and "vpn" are prioritized over a decent New Tab experience, which is basic UI job for a *browser*. I hope the team will reconsider; otherwise switching to Chromium-based alternatives like Kiwi or Samsung Browser, both of which with multiple New Tab options, will be our last resort.

**Comment by RowanSkie** [comment]
At least there's a timeframe. I use Kiwi now for Android, using Firefox and Brave once in a while but the stack view works really well for Android.

**Comment by elliedriod** [comment]
@timchilds @tapanmodh any update on this? haven't updated the app since March as i physically am unable to use the tab grid due to eyesight issues. this is an accessibility issue.

**Comment by theridon** [comment]
> @timchilds @tapanmodh any update on this? haven't updated the app since March as i physically am unable to use the tab grid due to eyesight issues. this is an accessibility issue.

They are not going to fix this. Just use Kiwi browser, it's just as good and has a proper tab layout.

**Comment by Sosasees** [comment]
i hope that i can keep the current grid tab view when they add back the old tab stack view.
because i find it much easier to see the thumbnails and reorder the tabs in the new grid tab view.

**Comment by 424344** [comment]
> keep the current grid tab view

yeah, I agree. please keep grid view.

**Comment by RowanSkie** [comment]
I mean, before the old got removed and news broke out, the new/current one was available by a switch.

**Comment by Enriador** [comment]
Devs already came out and said the plan is to have "multiple" options, which obviously includes the Chromium default (i.e. the awful cramp-inducing grid view). Heck, the issue literally starts with "Add", not "Replace".

There is no reason for @Sosasees or @424344 to reply random stuff like that - we get some folk like grid view, but that's been touched/solved already. Please read the entire conversation before posting next time.

**Comment by StrikerROSV** [comment]
Can we get an update on this, or the idea has died out?

**Comment by Enriador** [comment]
I have already switched to Kiwi where stacked tabs are one of many options, but it is fun to still get notifications on this and see nothing has been done.

I recall the devs promised on Reddit that it would take "weeks" to finish this up. That was in April 2022. 

So yeah I can confirm on their behalf that this idea is *de facto* dead and the issue didn't close just to shut people up with "we are working on it".

**Comment by thecrazypozavnist** [comment]
came here hoping to find a solution... So sad there's still no update after a year. Brave is great on desktop, but for my smartphone? I'll be switching to kiwi as multiple users pointed out I guess.

**Comment by true2thepen** [comment]
Brave just keeps on doing Stupid. I typically have 40 to 50 tabs. Scrolling through tabs in grid view is a waste of time. Guess Brave gets dumped. They've had more than enough time to fix this. Stacked tabs was there to begin with. Guess when you constantly refer to what Google is doing, you shoot yourself in the foot. Kiwi does indeed have stacked tabs. Goodbye Brave, Hello Kiwi.

**Comment by wegrata** [comment]
> @elliedriod `tab view mode` is a project on the 2023 roadmap. This will allow users to switch between different tab views, including a stack view. The project is currently in the design phase.

So is this just behind, inaccurate or just a lie?

**Comment by notvenom** [comment]
> @elliedriod `tab view mode` is a project on the 2023 roadmap. This will allow users to switch between different tab views, including a stack view. The project is currently in the design phase.

And of course, this was a lie.
Great work, remove a working feature for no reason, then give yourself a very wide timeframe (1.5 years) to return it, yet still fail to deliver on your promise or even give updates at all.
Pathetic.

**Comment by futurejohn** [comment]
> > @elliedriod `tab view mode` is a project on the 2023 roadmap. This will allow users to switch between different tab views, including a stack view. The project is currently in the design phase.
> 
> And of course, this was a lie.
> Great work, remove a working feature for no reason, then give yourself a very wide timeframe (1.5 years) to return it, yet still fail to deliver on your promise or even give updates at all.
> Pathetic.

Yep, just close this and say you're not going to do it, Brave. It's ok, there are other browsers for us to use.

**Comment by bsclifton** [comment]
Hi folks - thanks for your patience. I'm cleaning up some of the unhelpful comments here. While they do capture raw sentiment, I'd like to keep us on topic.  I do appreciate how it can be frustrating losing a feature like this.

For people not caught up:
- Chromium had this code originally
- Chromium removed it - but we were able to show it anyways
- The code was deleted from Chromium and we had no way to show this in Brave

@samartnik (Brave employee) had put together a working implementation after it was removed. The work was done with the following pull request in 2022 when this ticket was created:
https://github.com/brave/brave-core/pull/12684

However, this feature ("Stack View") never bubbled to the top priority-wise, compared to all the other issues we face and features we have on the roadmap. It came close to being staffed - but there was always something else. And here we are 1.5 years later.

We have a small number of developers on Android and putting back all of the code that Chromium deleted and then maintaining it long term is a difficult proposition. There were reasons Chromium removed it - related to performance and we had seen crashes related to this feature too. Setting that aside completely, having this extra code also increases the complexity of doing Chromium upgrades.

At the moment of this comment, work is no longer planned on this feature. We did on multiple occasions try to pull the work in but ultimately placed other work ahead of this.

**Comment by ArmandFrvr** [comment]
Don't want to clutter this up further but this is literally the one thing
preventing me from using Brave on my phone.  I don't see how it hasn't been
a priority.

**Comment by theridon** [comment]
Why am I getting notifications about this years later? Forget Brave, they aren't going to change it. Just switch over Kiwi Browser like I did.

**Comment by RowanSkie** [comment]
If by chance a random person like me wants to go on and work on this feature, what stops us? Is it the old code, or the crash problems caused by later android versions?

**Comment by bsclifton** [comment]
@RowanSkie if someone was interested in taking this on (perhaps starting with the original work in https://github.com/brave/brave-core/pull/12684) we can definitely help with reviewing and acceptance 😄 We do have some good documented best practices we can help share too - for example, with patching:
https://github.com/brave/brave-browser/wiki/Patching-Chromium

**Comment by bughit** [comment]
In the mean time Google continues making the tab switcher as useless as possible. Here's what it looks like now for tabs opened in the background (both Chrome and Brave):

![image](https://github.com/brave/brave-browser/assets/895588/f62dc895-2102-47bb-bbea-33a49a582c1c)

No more content previews, mostly empty space and the one useful piece of info that's actually available, the title, is mostly truncated.

And of course Brave continues inheriting this madness with no attempt to keep the browser usable.

**Comment by cyraid** [comment]
This sadly is the biggest thing making me want to move to a different browser. Opera has a nice tab switcher, and they base off Chromium? Opera's mobile browser is one of the best (IMO). One of the only reasons I'm using Brave is because I'm using Brave on the desktop and I'm used to it and I want it to sync to my mobile browser. How does Opera do it? Is it purely because of lack of manpower?

This issue is so upvoted, what is the other manpower doing then? As this is so highly upvoted issue. What are the main focuses? Can we vote on top issues for what the devs should work on and what matters to us? What are you guys working on?

**Comment by wegrata** [comment]
> This sadly is the biggest thing making me want to move to a different browser. Opera has a nice tab switcher, and they base off Chromium? Opera's mobile browser is one of the best (IMO). One of the only reasons I'm using Brave is because I'm using Brave on the desktop and I'm used to it and I want it to sync to my mobile browser. How does Opera do it? Is it purely because of lack of manpower?
> 
> This issue is so upvoted, what is the other manpower doing then? As this is so highly upvoted issue. What are the main focuses? Can we vote on top issues for what the devs should work on and what matters to us? What are you guys working on?

Dude if I thought there was half a chance of folks merging my pr I'd have written it. The grid view is unusable for me.
